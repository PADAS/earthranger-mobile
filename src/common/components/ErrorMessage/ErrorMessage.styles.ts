// External Dependencies
import { StyleSheet } from 'react-native';

// Internal Dependencies
import { COLORS_LIGHT } from '../../constants/colors';

const style = StyleSheet.create({
  errorMessage: {
    paddingLeft: 8,
    paddingRight: 8,
    paddingTop: 16,
    position: 'absolute',
    width: '100%',
    zIndex: 2,
  },
  errorMessageContainer: {
    alignItems: 'center',
    backgroundColor: COLORS_LIGHT.G0_black,
    opacity: 0.9,
    borderRadius: 3,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingBottom: 14,
    paddingLeft: 16,
    paddingRight: 16,
    paddingTop: 14,
  },
  errorMessageText: {
    color: COLORS_LIGHT.white,
    maxWidth: '90%',
  },
  errorMessageIcon: {
    marginLeft: 16,
  },
});

export default style;
