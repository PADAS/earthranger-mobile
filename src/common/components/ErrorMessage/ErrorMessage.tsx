/* eslint-disable arrow-body-style */
// External Dependencies
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Text, View, Pressable } from 'react-native';

// Internal Dependencies
import { CloseIcon } from '../../icons/CloseIcon';
import { COLORS_LIGHT } from '../../constants/colors';

// Styles
import style from './ErrorMessage.styles';

// Interfaces
interface ErrorMessageProps {
  onIconClickHandler: () => void;
}

export const ErrorMessage = ({ onIconClickHandler }: ErrorMessageProps) => {
  // Hooks
  const { t } = useTranslation();

  return (
    <View style={style.errorMessage}>
      <View style={style.errorMessageContainer}>
        <Text style={style.errorMessageText}>{t('statusView.unableToSync')}</Text>
        <Pressable style={style.errorMessageIcon} onPress={onIconClickHandler}>
          <CloseIcon color={COLORS_LIGHT.white} />
        </Pressable>
      </View>
    </View>
  );
};
