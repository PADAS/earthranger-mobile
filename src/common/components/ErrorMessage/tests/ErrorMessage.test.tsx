// External Dependencies
import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

// Internal Dependencies
import { ErrorMessage } from '../ErrorMessage';

describe('ErrorMessage', () => {
  it('matches snapshot', async () => {
    const { toJSON, getByText } = render(
      <ErrorMessage onIconClickHandler={jest.fn()} />,
    );

    await waitFor(() => {
      expect(getByText('Unable to sync, try again later.')).toBeTruthy();
    });

    expect(toJSON()).toMatchSnapshot();
  });
});
