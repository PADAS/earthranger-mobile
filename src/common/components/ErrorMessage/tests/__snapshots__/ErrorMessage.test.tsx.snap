// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage matches snapshot 1`] = `
<View
  style={
    Object {
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 16,
      "position": "absolute",
      "width": "100%",
      "zIndex": 2,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000000",
        "borderRadius": 3,
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 0.9,
        "paddingBottom": 14,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 14,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "maxWidth": "90%",
        }
      }
    >
      Unable to sync, try again later.
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginLeft": 16,
        }
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        focusable={false}
        height="16"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        vbHeight={16}
        vbWidth={16}
        width="16"
      >
        <RNSVGGroup>
          <RNSVGPath
            d="M16 1.61143L14.3886 0L8 6.38857L1.61143 0L0 1.61143L6.38857 8L0 14.3886L1.61143 16L8 9.61143L14.3886 16L16 14.3886L9.61143 8L16 1.61143Z"
            fill={4294967295}
            propList={
              Array [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
